buf/registry/plugin/v1beta1/plugin_service.proto

```diff
@@ -0,0 +1,194 @@
+  rpc DeletePlugins(DeletePluginsRequest) returns (DeletePluginsResponse) {
+    option idempotency_level = IDEMPOTENT;
+  }
+}
+
+message GetPluginsRequest {
+  // The Plugins to request.
+  repeated PluginRef plugin_refs = 1 [
+    (buf.validate.field).repeated.min_items = 1,
+    (buf.validate.field).repeated.max_items = 250
+  ];
+}
+
+message GetPluginsResponse {
+  // The retrieved Plugins in the same order as requested.
+  repeated Plugin plugins = 1 [(buf.validate.field).repeated.min_items = 1];
+}
+
+message ListPluginsRequest {
+  // The list order.
+  enum Order {
+    ORDER_UNSPECIFIED = 0;
+    // Order by create_time newest to oldest.
+    ORDER_CREATE_TIME_DESC = 1;
+    // Order by create_time oldest to newest.
+    ORDER_CREATE_TIME_ASC = 2;
+  }
+  // The maximum number of items to return.
+  //
+  // The default value is 10.
+  uint32 page_size = 1 [(buf.validate.field).uint32.lte = 250];
+  // The page to start from.
+  //
+  // If empty, the first page is returned.
+  string page_token = 2 [(buf.validate.field).string.max_len = 4096];
+  // The specific Organizations to list Plugins for.
+  //
+  // If empty, all Plugins for all Organizations are listed, but this functionality is limited to
+  // Users with the necessary permissions.
+  repeated buf.registry.owner.v1.OwnerRef owner_refs = 3;
+  // The collections to list Plugins for.
+  //
+  // If empty, all Plugins are listed regardless of collection.
+  repeated string collection_names = 4;
+  // The order to return the Plugins.
+  //
+  // If not specified, defaults to ORDER_CREATE_TIME_DESC.
+  Order order = 5 [(buf.validate.field).enum.defined_only = true];
+}
+
+message ListPluginsResponse {
+  // The next page token.
+  //
+  // If empty, there are no more pages.
+  string next_page_token = 1 [(buf.validate.field).string.max_len = 4096];
+  // The listed Plugins.
+  repeated Plugin plugins = 2;
+}
+
+message CreatePluginsRequest {
+  // An individual request to create a Plugin.
+  message Value {
+    // The Organization to create the Plugin under. The OwnerRef must be a reference to an
+    // Organization and cannot be a User reference.
+    buf.registry.owner.v1.OwnerRef owner_ref = 1 [(buf.validate.field).required = true];
+    // The name of the Plugin.
+    string name = 2 [(buf.validate.field).string = {
+      min_len: 2
+      max_len: 100
+    }];
+    // The plugin's visibility.
+    PluginVisibility visibility = 3 [
+      (buf.validate.field).required = true,
+      (buf.validate.field).enum.defined_only = true
+    ];
+    // The configurable description of the Plugin.
+    string description = 4 [(buf.validate.field).string.max_len = 350];
+    // The configurable source URL of the Plugin.
+    string source_url = 5 [
+      (buf.validate.field).string.uri = true,
+      (buf.validate.field).string.max_len = 255,
+      (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED
+    ];
+    // The Plugin type.
+    PluginType type = 6 [
```

Comment:
What does it mean to create a plugin of type `PLUGIN_TYPE_PROTOC`? More generally, what does it mean to create a plugin without any commits? How is (or how is this not) congruent with Modules?